src/shared_modules/utils/rsaHelper.hpp

```diff
@@ -56,12 +55,13 @@ class RSAHelper final
         // Defered free
         DEFER([&]() { T::RSA_free(rsa); });
 
+        // PKCS  #1  v1.5  padding.  This currently is the most widely used mode.  However, it is highly recommended to
+        // use RSA_PKCS1_OAEP_PADDING in new applications.
         const auto encryptedLen = T::RSA_public_encrypt(input.length(),
```

Comment:
This issue is still pending:
- https://github.com/wazuh/wazuh/pull/24529#discussion_r1672326487

We may want to have a table that represents this PADDING to max length relationships

Reply:
I don't understand the issue there.

Reply:
I will expand my comment

---

```C++
 int RSA_public_encrypt(int flen, const unsigned char *from,
                        unsigned char *to, RSA *rsa, int padding);

 int RSA_private_decrypt(int flen, const unsigned char *from,
                         unsigned char *to, RSA *rsa, int padding);
```

The documentation states that `flen` must not be bigger than `RSA_size(rsa) - k` (being `k` a variable depending on the padding type)

---

This line may catch the problem, but its logging is poor
```C++
        if (decryptedLen < 0)
        {
            throw std::runtime_error("RSA decryption failed");
        }
```

We could improve if we use the error functions from the library:
- [ERR_get_error](https://www.openssl.org/docs/man1.1.1/man3/ERR_get_error.html)
- [ERR_error_string](https://www.openssl.org/docs/man1.1.1/man3/ERR_error_string.html)

Reply:
The first message [here](https://github.com/wazuh/wazuh/pull/24529#discussion_r1672326487) mentioned some checks to the size provided to the OpenSSL primitives that I didn't consider necessary since everything is handled internally by the library, that's why I didn't understand the request. Now that everything is clear I added a minor change to clarify the error message, please take a look.  

This would've been the error prior to this fix

`2024/07/15 11:54:29 wazuh-modulesd:vulnerability-scanner[65921] vulnerabilityScannerFacade.cpp:484 at start(): ERROR: VulnerabilityScannerFacade::start: RSA decryption failed: oaep decoding error`